activesupport/lib/active_support/cache/file_store.rb

```diff
@@ -131,6 +131,7 @@ def normalize_key(key, options)
 
           # Make sure file name doesn't exceed file system limits.
           begin
+            fname = ActiveSupport::Digest.hexdigest(key) if EXCLUDED_DIRS.include?(fname)
```

Comment:
I think `.` and `..` are not the only, there are many another characters that invalid in mac, linux, or windows

Reply:
Invalid characters will have been escaped by `URI.encode_www_form_component`.
https://github.com/rails/rails/blob/master/activesupport/lib/active_support/cache/file_store.rb#L121

Dot isn't invalid character, but only single or double dot are invalid for filename.

Reply:
ok great
btw should we update old rails too? 
ActiveSupport::Digest is not available on Rails 4

Reply:
no need
in my company I use Rails 4, but never use File storage as cache
I think no need to create patch because no one created issue about this for rails 4